### PR TITLE
Fixing bugs

### PR DESCRIPTION
**Pull request by ikamman**
Hi I managed to fix some bugs which came up during using the tool.

Fix include:
* When using token authorisation the url for fetching the token is invalid. This will came up when your service name has spaces eg. "Docker Registry"  in the registry:2 image configuration.
* Tool does not follow redirects correctly. Credentials used in previous call are not copied to the next one.
* S3 support. During calling for blob date registry redirects to the s3 resource with tokens available in parameters. In that case authorisation header should be removed.

Hope this helps someone.

### CODE REVIEW

**Comment by eed3si9n** [comment]
@ikamman Thanks for the contribution.

File: cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala [hunk 1]
```diff
@@ -44,12 +46,22 @@ object Http {
...
-          if (visitedUrls.contains(location) || visitedUrls.length >= settings.maxRedirects)
+          if (visitedUrls.contains(location) || visitedUrls.length >= settings.maxRedirects) {
+            logger.debug("No more redirects allowed")
             Future.failed(InfiniteRedirect(visitedUrls))
-          else
-            doRequest(
-              HttpRequest(location).copy(requestFollowRedirects = request.requestFollowRedirects),
-              location :: visitedUrls)
+          } else {
+            println(location)
```
Comment (eed3si9n) [inline cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala]: Println is left here.

File: cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala [hunk 2]
```diff
@@ -44,12 +46,22 @@ object Http {
...
+            if (location.indexOf("X-Amz-Credential") > -1) {
```
Comment (eed3si9n) [inline cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala]: In general, could you split this change to another commit with a message describing what it does please?
What you wrote in the PR description is fine.

File: cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala [hunk 3]
```diff
@@ -44,12 +46,22 @@ object Http {
...
+              // S3 based registry doesn't accept authorization header on redirect
+              logger.debug("Performing S3 redirect")
+              doRequest(
+                HttpRequest(location, tlsValidationEnabled = Some(true), requestFollowRedirects = request.requestFollowRedirects),
+                location :: visitedUrls)
+            } else {
+              doRequest(
+                request.copy(location), location :: visitedUrls)
```
Comment (eed3si9n) [inline cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala]: Should this fall back to what the code is currently doing?

```
            doRequest(
              HttpRequest(location).copy(requestFollowRedirects = request.requestFollowRedirects),
              location :: visitedUrls)
 ```

Comment (ikamman) [reply cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala]: It shouldn't because the code

`doRequest(
              HttpRequest(location).copy(requestFollowRedirects = request.requestFollowRedirects),
              location :: visitedUrls)`

doesn't copy auth credentials which is required by a docker registry.

File: cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala (hunk 1 above)
Comment (ikamman) [reply cli/shared/src/main/scala/com/lightbend/rp/reactivecli/http/Http.scala]: Removed
